application-admintools-ui/src/main/resources/AdminTools/Code/Macros.xml

```diff
@@ -0,0 +1,134 @@
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+<xwikidoc version="1.4" reference="AdminTools.Code.Macros" locale="">
+  <web>AdminTools.Code</web>
+  <name>Macros</name>
+  <language/>
+  <defaultLanguage/>
+  <translation>0</translation>
+  <creator>xwiki:XWiki.Admin</creator>
+  <parent>AdminTools.WebHome</parent>
+  <author>xwiki:XWiki.Admin</author>
+  <contentAuthor>xwiki:XWiki.Admin</contentAuthor>
+  <version>1.1</version>
+  <title>Macros</title>
+  <comment/>
+  <minorEdit>false</minorEdit>
+  <syntaxId>xwiki/2.1</syntaxId>
+  <hidden>true</hidden>
+  <content>{{velocity output="false"}}
+#macro (viewLastNLinesMoldal $id)
+  #set ($modalId = $id + "ViewLastNLinesModal")
+  #set ($formId = $id + "LastNLogLinesForm")
+  #set ($inputId = $id + "NoLines")
+  {{html clean='false'}}
+    &lt;div class="modal fade" id="$modalId" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+      &lt;div class="modal-dialog modal-sm" role="document">
+        &lt;div class="modal-content">
+          &lt;div class="logs-modal-content">
+            &lt;div class="modal-header">
+              &lt;button type="button" class="close" data-dismiss="modal" aria-label="Close">
+              &lt;span aria-hidden="true">&amp;times;&lt;/span>&lt;/button>
+              &lt;h1 class="modal-title" id="viewLastNLinesModalLabel">
+                $escapetool.xml($services.localization.render('adminTools.dashboard.logs.modal.title'))
+              &lt;/h1>
+            &lt;/div>
+            &lt;div class="modal-body">
+              &lt;form class="xform" id="$formId" action="$request.getContextPath()/rest/admintools/files/logs">
+                &lt;dl>
+                  &lt;dt>
+                    &lt;label for="noLines">$escapetool.xml($services.localization.render('adminTools.dashboard.logs.modal.nLines.label')) &lt;/label>
+                    &lt;span class="xHint">max 50000&lt;/span>
+                  &lt;/dt>
+                  &lt;dd>&lt;input type="number" id="$inputId" name="noLines" placeholder="1000">&lt;/dd>
+                &lt;/dl>
+              &lt;/form>
+            &lt;/div>
+            &lt;div class="modal-footer">
+              &lt;button type="button" class="btn btn-primary">
+                $escapetool.xml($services.localization.render('adminTools.dashboard.logs.modal.submit'))&lt;/button>
+              &lt;button type="button" class="btn btn-default" data-dismiss="modal">
+                $escapetool.xml($services.localization.render('cancel'))&lt;/button>
+            &lt;/div>
+          &lt;/div>
+        &lt;/div>
+      &lt;/div>
+    &lt;/div>
+  {{/html}}
+#end
+
+#macro (downloadFiles $id)
```

Comment:
In the original design, the download option should appear in two sections inside the page (the files section and inside the table of contents). Although the table of contents it's not yet done, I considered it is reasonable to implement the id option for that purpose.

Reply:
I think that during the last meeting, this modal remained only as part of the Support tab, but it's added here (Files section) for now since the Support tab is not yet implemented

Reply:
I'd leave it like this for now and see if it needs to be removed when implementing the support tab. WDYT?

Reply:
Since it is supposed to be used only in one place (now Files, then moved to Support), I don't see why not adding it simplified (without the id param) from the start. You can refactor it in the future if the needed, but for now it is not planned